otx/algorithms/detection/adapters/mmdet/models/assigners/custom_max_iou_assigner.py

```diff
@@ -0,0 +1,107 @@
+    """
+
+    cpu_assign_thr = 1000
+
+    def assign(self, bboxes, gt_bboxes, gt_bboxes_ignore=None, gt_labels=None):
+        """Assign gt to bboxes.
+
+        This method assign a gt bbox to every bbox (proposal/anchor), each bbox
+        will be assigned with -1, or a semi-positive number. -1 means negative
+        sample, semi-positive number is the index (0-based) of assigned gt.
+        The assignment is done in following steps, the order matters.
+
+        Especially CustomMaxIoUAssigner split gt_bboxes tensor into small tensors
+        when gt_bboxes is too large.
+
+        1. assign every bbox to the background
+        2. assign proposals whose iou with all gts < neg_iou_thr to 0
+        3. for each bbox, if the iou with its nearest gt >= pos_iou_thr,
+           assign it to that bbox
+        4. for each gt bbox, assign its nearest proposals (may be more than
+           one) to itself
+
+        Args:
+            bboxes (Tensor): Bounding boxes to be assigned, shape(n, 4).
+            gt_bboxes (Tensor): Groundtruth boxes, shape (k, 4).
+            gt_bboxes_ignore (Tensor, optional): Ground truth bboxes that are
+                labelled as `ignored`, e.g., crowd boxes in COCO.
+            gt_labels (Tensor, optional): Label of gt_bboxes, shape (k, ).
+
+        Returns:
+            :obj:`AssignResult`: The assign result.
+
+        Example:
+            >>> self = MaxIoUAssigner(0.5, 0.5)
+            >>> bboxes = torch.Tensor([[0, 0, 10, 10], [10, 10, 20, 20]])
+            >>> gt_bboxes = torch.Tensor([[0, 0, 10, 9]])
+            >>> assign_result = self.assign(bboxes, gt_bboxes)
+            >>> expected_gt_inds = torch.LongTensor([1, 0])
+            >>> assert torch.all(assign_result.gt_inds == expected_gt_inds)
+        """
+        assign_on_cpu = True if (self.gpu_assign_thr > 0) and (gt_bboxes.shape[0] > self.gpu_assign_thr) else False
+        # compute overlap and assign gt on CPU when number of GT is large
+        if assign_on_cpu:
+            device = bboxes.device
+            bboxes = bboxes.cpu()
+            gt_bboxes = gt_bboxes.cpu()
+            if gt_bboxes_ignore is not None:
+                gt_bboxes_ignore = gt_bboxes_ignore.cpu()
+            if gt_labels is not None:
+                gt_labels = gt_labels.cpu()
+
+        if assign_on_cpu and gt_bboxes.shape[0] > self.cpu_assign_thr:
+            split_length = gt_bboxes.shape[0] // self.cpu_assign_thr + 1
+            overlaps = None
+            for i in range(split_length):
+                gt_bboxes_split = gt_bboxes[i * self.cpu_assign_thr : (i + 1) * self.cpu_assign_thr]
+                if overlaps is None:
+                    overlaps = self.iou_calculator(gt_bboxes_split, bboxes)
+                else:
+                    overlaps = torch.concat((overlaps, self.iou_calculator(gt_bboxes_split, bboxes)), dim=0)
+
+        else:
+            overlaps = self.iou_calculator(gt_bboxes, bboxes)
```

Comment:
The size of `overlaps` tensor is same regardless of looping. Then, does it mean that some internal buffers used in `self.iou_calculater()` had been causing the CPU OOM issue?

Additionally,
```suggestion
        if assign_on_cpu and gt_bboxes.shape[0] > self.cpu_assign_thr:
            split_length = gt_bboxes.shape[0] // self.cpu_assign_thr + 1
            overlaps = []
            for i in range(split_length):
                gt_bboxes_split = gt_bboxes[i * self.cpu_assign_thr : (i + 1) * self.cpu_assign_thr]
                overlaps.append(self.iou_calculator(gt_bboxes_split, bboxes))
            overlaps = torch.concat(overlaps, dim=0)
        else:
            overlaps = self.iou_calculator(gt_bboxes, bboxes)
```

Reply:
For your question, yes self.iou_calculator generates (9 x N x M + N + M) matrix, where N is number of gt and M is number of anchor, to return NxM output matrix. Therefore you can think this may prevent OOM from 9 x N x M + N + M matrix but can't prevent from very large N x M matrix